sw/device/examples/sram_program/BUILD

```diff
@@ -42,23 +42,55 @@ opentitan_gdb_fpga_cw310_test(
     exit_success_pattern = "sram_program\\.c:47\\] PC: 0x1000208c, SRAM: \\[0x10000000, 0x10020000\\)",
     gdb_script = """
         target extended-remote :3333
-        set pagination off
 
         echo :::: Send OpenOCD the 'reset halt' command.\\n
         monitor reset halt
 
-        echo :::: Set RAM as RWX by executing code between _start and test_rom_main()\\n
-        file test_rom.elf
-        break rom_test_main
+        # Disable the watchdog timer as quickly as possible. If it fires, the
+        # exception will steal control from the debugger. If we're unlucky, the
+        # watchdog timer might fire before the commands below have disabled it,
+        # causing this test to fail. The proper solution would be to disable ROM
+        # execution by zeroing out CREATOR_SW_CFG_ROM_EXEC_EN in the OTP.
+        file rom.elf
+        break sec_mmio_init
         continue
+        finish
+        print watchdog_disable()
         delete breakpoints
 
+        # Before transferring the SRAM program to the device, we must configure
+        # the PMP unit to enable writing to and executing from SRAM. Due to
+        # implementation details of OpenTitan's hardware debug module, we cannot
+        # set pmpcfg* registers to arbitrary values[1]. However, we can safely
+        # modify unused PMP configuration registers. Thankfully, pmp0cfg (the
+        # lowest byte of CSR pmpcfg0) is unused and has the highest priority.
+        #
+        # In more detail, the problem is that our debug module implements the
+        # "Access Register" abstract command by assembling instructions in the
+        # program buffer and then executing the buffer. If one of those
+        # instructions clobbers the PMP configuration register that allows
+        # execution from the program buffer, subsequent instruction fetches will
+        # generate exceptions. To work around this, we simply leave any in-use
+        # PMP configuration registers alone. Debug module concepts like abstract
+        # commands and the program buffer buffer are defined in "RISC-V External
+        # Debug Support Version 0.13.2"[2]. OpenTitan's (vendored-in)
+        # implementation lives in //hw/vendor/pulp_riscv_dbg.
+        #
+        # [1]: https://github.com/lowRISC/opentitan/issues/14978
+        # [2]: https://riscv.org/wp-content/uploads/2019/03/riscv-debug-release.pdf
+
+        echo :::: Configure the PMP unit.\\n
+        # Print the value of pmpcfg0 before setting it to demonstrate that we
+        # preserve the values of pmp{1,2,3}cfg. (Requires manual inspection.)
+        monitor reg pmpcfg0
+        monitor reg pmpcfg0 0x009d009f
```

Comment:
What happens if we write 0x9f9f9f9f, i.e. L NAPOT RWX cfg fields 0-3?

Reply:
Confirmed, that works, too. Would you prefer that over `0x009d009f`?

Reply:
Yes, please. This way, we can also remove the comment about other pmp regions, right?

Reply:
Yeah, I'm rewriting the comment and removing the part that calls for manual inspection.